docs/api_documentation.md

```diff
@@ -0,0 +1,36 @@
+# API Docs
+## Generating Docs
+Docs are autogenerated based off of the requests specs, no extra code is needed.
+
+The specs are run with the OPENAPI env var set to trigger the doc generation, it is set on pushes only in the `main` branch and when changes are made to `spec/requests/api`.
+
+If you have branch protections make sure to set the `PUSH_KEY` secret so the CI can update the docs.
```

Comment:
now that we adapted the workflow, isn't it always necessary? or does the workflow works without the `PUSH_KEY`?

Reply:
why did we need this for this job and not the others?

Reply:
I think you're right since this is a push to the repo